structuredevent-model/src/main/java/com/sequenceiq/cloudbreak/structuredevent/event/cdp/CDPStructuredFlowEvent.java

```diff
@@ -0,0 +1,73 @@
+package com.sequenceiq.cloudbreak.structuredevent.event.cdp;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.sequenceiq.cloudbreak.structuredevent.event.FlowDetails;
+import com.sequenceiq.cloudbreak.structuredevent.json.Base64Deserializer;
+import com.sequenceiq.cloudbreak.structuredevent.json.Base64Serializer;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CDPStructuredFlowEvent<T> extends CDPStructuredEvent {
```

Comment:
I think `T` should be serializable

Reply:
I don't want to add the `Serializable` to the generic type. Sometimes generate unnecessary changes/refactors.

Reply:
but `CDPStructuredEvent` is serializable and a class is only serializable if all the fields are serializable or transient. In this case you don't have to modify anything else as I have checked and the constructor is used only at one place where payload is set to `null`